drivers/i2c/i2c_gpio.c

```diff
@@ -47,6 +47,7 @@ struct i2c_gpio_config {
 /* Driver instance data */
 struct i2c_gpio_context {
 	struct i2c_bitbang bitbang;	/* Bit-bang library data */
+	struct k_mutex mutex;
```

Comment:
as far as I know, mutexes are designed for threads. However, can we be sure that changed functions won't be invoked from an ISR?

Reply:
you shouldn't be calling a 'bit-banged' driver function that is gonna take a long time from the beginning, that would be a very bad ISR as that could take a long time.

Reply:
so, should we add a note about this somewhere? "shouldn't" -- it is good, but my question was - can we be sure about that?
what about calls with IRQs disabled, e.g. atomic context?

Reply:
Then, do you mean to prevent IRQs from occurring while using I2C GPIO?

Reply:
The mutex api already has asserts that check if it is within an ISR. Anything outside of that is redundant and unnecessary. Every place that I've seen use a muted does not need an additional comment or code check as those should understand that a muted already is. 

The code looks good as is... but you gotta fix the compliance issues with the git commit and push again.

Reply:
I fixed the compliance issue with git commit and pushed it again.

Reply:
got it, thank you